packages/jest-haste-map/src/crawlers/node.ts

```diff
@@ -25,13 +24,29 @@ type Callback = (result: Result) => void;
 async function hasNativeFindSupport(
   forceNodeFilesystemAPI: boolean,
 ): Promise<boolean> {
-  if (forceNodeFilesystemAPI || process.platform === 'win32') {
+  if (forceNodeFilesystemAPI) {
     return false;
   }
 
   try {
-    await which('find');
-    return true;
+    return await new Promise(resolve => {
+      // Check the find binary supports the non-POSIX -iname parameter.
+      const args = ['.', '-iname', "''"];
+      const child = spawn('find', args);
+      let resolved = false;
+      child.on('error', () => {
+        if (!resolved) {
```

Comment:
do we need this `if`? Won't any subsequent calls to `resolve` just be ignored?

Reply:
Ah right, I was guarding against both events being triggered (as warned in https://nodejs.org/api/child_process.html#child_process_event_error) but didn't realize a second call to `resolve` would be ignored.

Reply:
Now removed.